packages/wonder-blocks-timing/src/hooks/use-interval.stories.mdx

```diff
@@ -0,0 +1,147 @@
+import {Meta, Story, Source, Canvas} from "@storybook/addon-docs";
+
+import {Body, HeadingSmall} from "@khanacademy/wonder-blocks-typography";
+import {View} from "@khanacademy/wonder-blocks-core";
+import Button from "@khanacademy/wonder-blocks-button";
+
+import {ClearPolicy, SchedulePolicy} from "../util/policies.js";
+import {useInterval} from "./use-interval.js";
+
+<Meta
+    title="Timing/useInterval"
+    parameters={{
+        chromatic: {
+            disableSnapshot: true,
+        },
+    }}
+/>
+
+# `useInterval`
+
+`useInterval` is a hook that provides a convenient API for setting and clearing
+an interval. It is defined as follows:
+
+```ts
+function useInterval(
+    action: () => mixed,
+    timeoutMs: number,
+    options?: {|
+        schedulePolicy?: "schedule-immediately" | "schedule-on-demand",
+        clearPolicy?: "resolve-on-clear" | "cancel-on-clear",
+    |},
+): ITimeout;
+
+interface ITimeout {
+    get isSet(): boolean;
+    set(): void;
+    clear(policy?: ClearPolicy): void;
+}
+```
+
+By default the interval will be set immediately up creation. The `options` parameter can
+be used to control when when the interval is schedule and whether or not `action` should be
+called when the interval is cleared.
+
+Notes:
+
+-   Because `clear` takes a param, it's import that you don't pass it directly to an event handler,
+    e.g. `<Button onClick={clear} />` will not work as expected.
+-   Calling `set` after the interval has been cleared will restart the interval.
+-   Updating the second paramter, `timeoutMs`, will also restart the interval.
+-   When the component using this hooks is unmounted, the interval will automatically be cleared.
+-   Calling `set` after the interval is already set does nothing.
+
+export const Immediately = () => {
+    const [callCount, setCallCount] = React.useState(0);
+    const callback = React.useCallback(() => {
+        setCallCount((callCount) => callCount + 1);
+    }, []);
+    const {isSet, set, clear} = useInterval(callback, 1000);
+    return (
+        <View>
+            <View>isSet = {isSet.toString()}</View>
+            <View>callCount = {callCount}</View>
+            <View style={{flexDirection: "row"}}>
+                <Button onClick={set}>Set interval</Button>
+                <Button onClick={clear}>Clear interval</Button>
```

Comment:
is this supposed to be a counter-example? because above it says "don't pass clear to onClick"

Reply:
This is me forgetting my own guidance.  Thankfully flow picks this up (I just checked).  The reason why Flow didn't pick this up here is that it doesn't run in .mdx files.